src/renderer/wg_engine/tvgWgCompositor.cpp

```diff
@@ -268,16 +275,22 @@ void WgCompositor::composeShape(WgContext& context, WgRenderDataShape* renderDat
     beginRenderPass(commandEncoder, &storageInterm, true);
     drawShape(context, renderData, WgPipelineBlendType::Custom);
     endRenderPass();
-    // run compisition
-    compose(commandEncoder, &storageInterm, mask, target, composeMethod);
     // restore current render pass
     beginRenderPass(commandEncoder, target, false);
+    // apply mask
+    wgpuRenderPassEncoderSetScissorRect(renderPassEncoder,0, 0, width, height);
```

Comment:
This code snippet seems duplicated four times. It could be refactored into a common function for reuse.
```
WgCompositor::applyMask(...)
{
    wgpuRenderPassEncoderSetStencilReference(renderPassEncoder, 0);
    wgpuRenderPassEncoderSetBindGroup(renderPassEncoder, 0, storageInterm.bindGroupTexure, 0, nullptr);
    wgpuRenderPassEncoderSetBindGroup(renderPassEncoder, 1, mask->bindGroupTexure, 0, nullptr);
    wgpuRenderPassEncoderSetPipeline(renderPassEncoder, pipelines->sceneComp[(uint32_t)composeMethod]);
    meshData.drawImage(context, renderPassEncoder);
}
```

Reply:
Make sence, Fixed